xchange-binance/src/main/java/org/knowm/xchange/binance/service/BinanceBaseService.java

```diff
@@ -35,30 +37,34 @@ protected BinanceBaseService(
         ExchangeRestProxyBuilder.forInterface(
                 BinanceAuthenticated.class, exchange.getExchangeSpecification())
             .build();
-    ExchangeSpecification futuresSpec = exchange.getDefaultExchangeSpecification();
-    ExchangeSpecification inverseFuturesSpec = futuresSpec;
-    futuresSpec.setSslUri(
-        (exchange.usingSandbox())
-            ? BinanceExchange.SANDBOX_FUTURES_URL
-            : (exchange.isPortfolioMarginEnabled())
-                ? BinanceExchange.PORTFOLIO_MARGIN_URL
-                : BinanceExchange.FUTURES_URL);
-    if (!exchange.isPortfolioMarginEnabled()) {
-      inverseFuturesSpec.setSslUri(
-          (exchange.usingSandbox())
-              ? BinanceExchange.SANDBOX_FUTURES_URL
-              : BinanceExchange.INVERSE_FUTURES_URL);
-      this.inverseBinanceFutures =
-          ExchangeRestProxyBuilder.forInterface(
-                  BinanceFuturesAuthenticated.class, inverseFuturesSpec)
-              .build();
-    } else {
-      this.inverseBinanceFutures = null;
+    ExchangeSpecification futuresSpec;
+    ExchangeSpecification inverseFuturesSpec;
+    if(exchange.getExchangeSpecification().getExchangeSpecificParametersItem(EXCHANGE_TYPE) != null)
+    {
+      switch ((ExchangeType) exchange.getExchangeSpecification()
+          .getExchangeSpecificParametersItem(EXCHANGE_TYPE)) {
+        case SPOT: {
+          break;
+        }
+        case FUTURES: {
+          futuresSpec = exchange.getExchangeSpecification();
+          binanceFutures =
```

Comment:
imho, you can use here single `ExchangeSpecification` variable and also single `BinanceFuturesAuthenticated` class member

Reply:
Yes, I noticed this.
but haven't bothered to fix it yet.
firstly, I’m not sure that the patch will be accepted,
secondly, I didn’t write this code, maybe some nuances will be revealed there